src/plugins/intel_gpu/src/plugin/plugin.cpp

```diff
@@ -970,21 +971,35 @@ Parameter Plugin::GetMetric(const std::string& name, const std::map<std::string,
         std::vector<ov::PropertyName> cachingProperties;
         cachingProperties.push_back(ov::PropertyName(uarch_version.name(), PropertyMutability::RO));
         cachingProperties.push_back(ov::PropertyName(execution_units_count.name(), PropertyMutability::RO));
-        cachingProperties.push_back(ov::PropertyName("GPU_DRIVER_VERSION", PropertyMutability::RO));
+        cachingProperties.push_back(ov::PropertyName(driver_version.name(), PropertyMutability::RO));
         return decltype(ov::caching_properties)::value_type(cachingProperties);
-    } else if (name == "GPU_DRIVER_VERSION") {
-        return device_info.driver_version;
+    } else if (name == ov::intel_gpu::driver_version) {
+        return decltype(ov::intel_gpu::driver_version)::value_type {device_info.driver_version};
     } else if (name == ov::device::architecture) {
-        std::stringstream s;
-        s << "GPU.";
-        if (device_info.gfx_ver.major == 0 && device_info.gfx_ver.minor == 0) {
-            s << device_info.dev_name;
+        // Ref: https://dgpu-docs.intel.com/devices/hardware-table.html
+        std::set<uint32_t> xehpg_set{0x56};
+        std::set<uint32_t>    xe_set{0x49, 0x46, 0x4C, 0x9A};
+        std::set<uint32_t> gen11_set{0x8A};
+        std::set<uint32_t>  gen9_set{0x3E, 0x9B, 0x87, 0x59, 0x31, 0x1A, 0x5A, 0x0A, 0x19};
+        std::set<uint32_t>  gen8_set{0x16, 0x22};
```

Comment:
Why have you changed to this approach instead of `gfx_ver`? New one is not future proof.

Reply:
I reverted to use `gfx_ver`, and to use ':' for separator. Now it outputs like 'GPU: v12.0.0'.